test/PProf.jl

```diff
@@ -78,4 +79,18 @@ end
     @test load_prof_proto(pprof(out=tempname(), web=false, keep_frames = "foo")).keep_frames != 0
 end
 
+@testset "@pprof macro" begin
+    let arr = []
+        @pprof foo(10000, 5, arr)
+        sleep(2)
+    end
+    @test PProf.proc[] !== nothing
+
+    PProf.kill()
+    @test PProf.proc[] === nothing
+
+    @test isfile("profile.pb.gz")
+    rm("profile.pb.gz")
+end
+
 end
```

Comment:
@gdkrmr  Can you please also add a couple tests for the new `refresh()` and `kill()` behavior? Maybe just verify that the old process is in fact killed after each function is called?

Something like:
```julia
current_proc = PProf.proc[]
@assert isopen(current_proc)
PProf.refresh()
@test !isopen(current_proc)
@test isopen(PProf.proc[])

current_proc = PProf.proc[]
@assert isopen(current_proc)
PProf.kill()
@test !isopen(current_proc)
@test !isopen(PProf.proc[])
```

Reply:
`isopen` does not really work for processes, I used `process_running` and `process_exited`.